tests/SpecTests/RetryableWritesSpecTest.php

```diff
@@ -29,15 +28,13 @@ public function testRetryableWrites(stdClass $test, array $runOn = null, array $
             $this->markTestSkipped('Transaction numbers are only allowed on a replica set member or mongos (PHPC-1415)');
         }
 
-        if (isset($test->useMultipleMongoses) && $test->useMultipleMongoses && $this->isShardedCluster()) {
-            $this->manager = new Manager(static::getUri(true));
```

Comment:
I realised way too late that we don't use `$this->manager` in these tests. This was exposed when testing session pinning, as running with a single mongos won't trigger pinning logic in libmongoc.

Reply:
Had we used `$this->manager`, which is set from `MongoDB\Tests\FunctionalTestCase::setUp()`, I don't think that would have worked either. That manager is constructed purely from the URI environment variable and wouldn't have considered the spec tests' `useMultipleMongoses` option.

Reply:
That's why I originally added this conditional to recreate the manager with multiple mongooses if specified in the spec test, not realising that it still didn't have any effect.